### PR TITLE
Add Websockets Support

### PR DESCRIPTION
**Pull request by iainbryson**
Very very basic websockets support.  I avoided refactoring (and the cost of some failure to DRY).  There are many ways for this to be improved, but I'd prefer to get the pull discussion started sooner rather than later so I can evolve it in the right direction.

### CODE REVIEW

File: lib/testserver.js [hunk 1]
```diff
@@ -57,6 +58,11 @@ function TestServer(options)
 		else
 		{
 			server = http.createServer(listen);
+
+			wsServer = new WebSocketServer({
+    		httpServer: server,
```
Comment (alexfernandez) [inline lib/testserver.js]: Spacing is a bit messy here, please use tabs throughout.

File: package.json [hunk 2]
```diff
@@ -15,7 +15,8 @@
 		"agentkeepalive": "^2.0.3",
 		"log": "1.4.*",
 		"stdio": "^0.2.3",
-		"testing": "*"
+		"testing": "*",
```
Comment (alexfernandez) [inline package.json]: Please add the websocket library above so you only have to change one line.

File: package.json [hunk 3]
```diff
@@ -15,7 +15,8 @@
...
+		"websocket": "*"
```
Comment (alexfernandez) [inline package.json]: IME it is better to depend on a specific version of each library, with the caret operator `^` in front. So it doesn't crash on any updates.

**Comment by alexfernandez** [comment]
Looks good, just a couple of minor issues.

**Comment by iainbryson** [comment]
Great, I think I've fixed all these issues.

**Comment by alexfernandez** [comment]
I am merging this. It would be great to document the new capabilities in the README file.

**Pull request merged**